File: 063.LeetCode@19_Remove_Nth_Node_From_End_of_List/063.LeetCode@19_Remove_Nth_Node_From_End_of_List.java
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int size=0;
        ListNode temp=head;
        while(temp!=null){
            size++;
            temp=temp.next;
        }
        if(size==n){
            head=head.next;
            return head;
        }
        ListNode prev=head;
        int i=1;
        while(i<(size-n)){
            prev=prev.next;
            i++;
        }
        prev.next=prev.next.next;
        return head;
    }
}

File: 063.LeetCode@19_Remove_Nth_Node_From_End_of_List/063.LeetCode@19_Remove_Nth_Node_From_End_of_List.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        size = 0
        temp = head
        while temp:
            size += 1
            temp = temp.next
        if size == n:
            return head.next
        prev = head
        for _ in range(1, size - n):
            prev = prev.next
        prev.next = prev.next.next

        return head